tests/common/monitor.py

```
@@ -0,0 +1,126 @@
+
+    def start(self, daemon_pid: int) -> None:
+        if not self.enabled or self._proc is not None:
+            return
+        self.out_dir.mkdir(parents=True, exist_ok=True)
+        self._marks = open(self.marks_path, "w", buffering=1)
+        self._write_mark("monitor_started", pid=daemon_pid)
+
+        # Run the sampler as a module so its imports resolve from the repo.
+        env = os.environ.copy()
+        env["PYTHONUNBUFFERED"] = "1"
+        self._proc = subprocess.Popen(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)